### PR TITLE
Associate explorations with schemas at API layer

### LINKED ISSUE #3880: Modify API to make explorations aware of their associated schema
## Context

1. In the UI, each exploration is associated with one schema by virtue of being presented "within" a schema.
1. In Mathesar 0.1.7, that association was handled by virtue of each `Exploration` referring to a `Table` which then referred to a `Schema`. All of those associations were contained within the internal database.
1. Mathesar 0.2 onwards no longer has that `Table` model. Each exploration weakly references a table oid in the user database. And we need to look inside the user database to see what schema that table lives within.
1. The change in data model presents a small problem, given our current UX.
1. At some meeting a few months ago we had decided to solve this problem within the front end. The front end would fetch all the explorations for a database and then filter them out.
1. However, now that I'm in the code I can see that taking that approach will require significant front end refactoring. Pavish and I had a [call](https://tldv.io/app/meetings/66f4095a1812d600131d14e2) about this today and decided that it would be best to solve this problem in the service layer instead of in the front end.

## Changes needed

Here's what we'd like:

- `explorations.list`

    - Add an optional `schema_oid` parameter to filter the list of explorations by schema.

    - Modify return value to include a `schema_oid` property.

- `explorations.get`

    - Modify return value to include a `schema_oid` property.

- `explorations.add`

    - Modify return value to include a `schema_oid` property.

To handle these changes, the service layer would need to call out to the user database for these two RPC methods.

### PR DESCRIPTION
**Pull request by seancolsen**
Fixes #3880

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by seancolsen** [comment]
@Anish9901 this is stacked on #3855. Please review but don't merge if that PR is still open.

File: mathesar/rpc/explorations.py [hunk 1]
```diff
@@ -6,6 +6,8 @@
 from modernrpc.core import rpc_method, REQUEST_KEY
 from modernrpc.auth.basic import http_basic_auth_login_required
 
+from db.connection import exec_msar_func
+
```
Comment (Anish9901) [inline mathesar/rpc/explorations.py:10]: We never import `exec_msar_func` in the endpoint definition files. Please create appropriate function under the `db/` namespace and use them here.

**Review by Anish9901 — CHANGES_REQUESTED** [review]
- I think it would be better if we stored `schema_oid` in the `Explorations` model instead of requesting the `schema_oid` everytime, especially since we don't allow moving tables within schemas through mathesar. This would also make listing and filtering much easier.

- If you do feel strongly about the way you've implemented it then, I'd prefer to getting the schema_oid in the `from_model` method for `ExplorationInfo` instead, you'd do this by 
   - Defining `get_relation_namespace_oid` function in `db/tables/operations/select.py`.
    - Altering `from_model` to take `conn` as an argument.
    - Calling `get_relation_namespace_oid` defined in select.py for the `schema_oid` field.

**Comment by seancolsen** [comment]
Abandoning in favor of #3892

**Pull request closed without merging**